client/src/portal/api/mod.rs

```diff
@@ -259,6 +259,22 @@ impl Keyring {
         )
     }
 
+    pub(crate) fn verify_secret(&self, secret: &Secret) -> bool {
+        let key = self.derive_key(secret);
+
+        // if this is a new keyring or a keyring with zero items
+        if self.items.is_empty() {
+            return true;
+        }
+
+        // attempting to decrypt the first item
+        if self.items[0].clone().decrypt(&key).is_ok() {
```

Comment:
Should this really call decrypt? Is trying to decrypt one item enough?

Reply:
> Should this really call decrypt?

yes, I think so. I don't see any other way to verify the keyring secret.

> Is trying to decrypt one item enough?

yes, I think so. all secrets (items) are encrypted using the same key.

cc @ueno

Reply:
I stumbled upon this as well; while it might be out of scope of our threat model, it is possible that the first item is corrupted. Maybe good to decrypt all items or at least have a comment explaining why decrypting the first item is sufficient.

Reply:
understood, thanks Daiki!